build-support/gen_build_version.py

```diff
@@ -54,6 +54,17 @@ def get_hostname():
     out, err = res.communicate()
     return out.decode('utf-8').strip()
 
+def get_build_distro_id():
+    os_release_path = '/etc/os-release'
+    build_distro_id = None
+    if os.path.exists(os_release_path):
```

Comment:
Do we need to output `ubuntu-22.04` and `gcc/clang` version instead of `ubuntu`?

Reply:
this PR is to add the distribution id to the binary version info, so it will be much easier to check if the user's running binary and os matches each other.

gcc/clang version can be obtained from the elf info which is not so obvious for now.